Makefile

```diff
@@ -41,6 +43,12 @@ fmt:
 vet:
 	go vet ./...
 
+unit:
```

Comment:
should we add the `mockgen` target here as a dependency?

Reply:
I might have it misconfigured, but AFAICT with `go generate` the mocks are recreated every time which adds about 20 seconds to running tests. 

I imagine that time will increase, so I would lean towards it being a one-off as necessary